src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPoolMetrics.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+namespace System.Buffers;
+
+internal sealed class PinnedBlockMemoryPoolMetrics
+{
+    // Note: Dot separated instead of dash.
+    public const string MeterName = "System.Buffers.PinnedBlockMemoryPool";
+
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentMemory;
+    private readonly Counter<long> _totalAllocatedMemory;
+    private readonly Counter<long> _evictedMemory;
+    private readonly Counter<long> _usageRate;
+
+    public PinnedBlockMemoryPoolMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.Create(MeterName);
+
+        _currentMemory = _meter.CreateUpDownCounter<long>(
+            "pinnedblockmemorypool.current_memory",
```

Comment:
This name should probably be something like `aspnetcore.memorypool.current_memory`. So use `aspnetcore.memorypool.*` as the prefix for values.